src/v/cluster/feature_manager.cc

```diff
@@ -288,7 +290,64 @@ ss::future<> feature_manager::maybe_log_license_check_info() {
     }
 }
 
+bool feature_manager::need_to_verify_enterprise_license() {
+    return features::is_major_version_upgrade(
+      _feature_table.local().get_active_version(),
+      _feature_table.local().get_latest_logical_version());
+}
+
+void feature_manager::verify_enterprise_license() {
+    vlog(clusterlog.info, "Verifying enterprise license...");
+
+    if (!need_to_verify_enterprise_license()) {
+        vlog(clusterlog.info, "Enterprise license verification skipped...");
+        _verified_enterprise_license = true;
+        return;
+    }
+
+    const auto& license = _feature_table.local().get_license();
+    auto license_missing_or_expired = !license || license->is_expired();
+    auto enterprise_features_used = license_required_feature_enabled();
+
+    vlog(
+      clusterlog.info,
+      "Verifying enterprise license: active_version={}, latest_version={}, "
+      "enterprise_features_used={}, license_missing_or_expired={}",
+      _feature_table.local().get_active_version(),
+      _feature_table.local().get_latest_logical_version(),
+      enterprise_features_used,
+      license_missing_or_expired);
+
+    if (enterprise_features_used && license_missing_or_expired) {
+        throw std::runtime_error{
+          "Looks like you’ve enabled a Redpanda Enterprise feature(s) without "
+          "a valid license. You cannot upgrade the cluster with Redpanda "
+          "Enterprise features without a valid license. Please downgrade this "
+          "broker to the pre-upgrade version and enter an active Redpanda "
+          "license key (e.g. rpk cluster license set <key>) before performing "
+          "the upgrade. If you don’t have one, please request a new/trial "
+          "license at https://redpanda.com/license-request"};
```

Comment:
@redpanda-data/documentation This is the error message I would like your review on for this PR. This is what we are going to print in the broker logs when the broker aborts upon detecting that the broker is undergoing a major version upgrade in a non-compliant enterprise license state.

Reply:
cc @mattschumpert

Reply:
Note: Github doesn't update the preview shown here, so make sure to click on the name of the file at the top of this comment and view the up-to-date message.

With the current state of the code, the message looks like this:
```
ERROR 2024-09-27 10:44:49,704 [shard 0:main] main - application.cc:511 - Failure during startup: std::runtime_error (Looks like you’ve enabled Redpanda Enterprise features ([rbac]) without a valid license. You cannot upgrade the cluster with Redpanda Enterprise features without a valid license. Please downgrade this broker to the pre-upgrade version and enter an active Redpanda license key (e.g. rpk cluster license set <key>) before performing the upgrade. If you don’t have one, please request a new/trial license at https://redpanda.com/license-request)
```